pkg/plugins/backendplugin/coreplugin/core_plugin.go

```diff
@@ -70,50 +71,50 @@ func (cp *corePlugin) Target() backendplugin.Target {
 }
 
 func (cp *corePlugin) CollectMetrics(_ context.Context, _ *backend.CollectMetricsRequest) (*backend.CollectMetricsResult, error) {
-	return nil, backendplugin.ErrMethodNotImplemented
+	return nil, plugins.ErrMethodNotImplemented
 }
 
 func (cp *corePlugin) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
 	if cp.CheckHealthHandler != nil {
 		return cp.CheckHealthHandler.CheckHealth(ctx, req)
 	}
 
-	return nil, backendplugin.ErrMethodNotImplemented
+	return nil, plugins.ErrMethodNotImplemented
 }
 
 func (cp *corePlugin) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
 	if cp.QueryDataHandler != nil {
 		return cp.QueryDataHandler.QueryData(ctx, req)
```

Comment:
Should this and other methods now return `plugins.ErrPluginDownstreamError` for consistency?

Reply:
I guess that would change the current behavior, do we want to do that?